samples/extensions/dynamic_rendering/dynamic_rendering.cpp

```diff
@@ -0,0 +1,481 @@
+			vkCmdBindDescriptorSets(draw_cmd_buffer, VK_PIPELINE_BIND_POINT_GRAPHICS, pipeline_layout, 0, 1, &descriptor_set, 0, nullptr);
+
+			// skybox
+			vkCmdBindPipeline(draw_cmd_buffer, VK_PIPELINE_BIND_POINT_GRAPHICS, skybox_pipeline);
+			draw_model(skybox, draw_cmd_buffer);
+
+			// object
+			vkCmdBindPipeline(draw_cmd_buffer, VK_PIPELINE_BIND_POINT_GRAPHICS, model_pipeline);
+			draw_model(object, draw_cmd_buffer);
+
+			// UI
+			draw_ui(draw_cmd_buffer);
+		};
+
+		VkImageSubresourceRange range{};
+		range.aspectMask     = VK_IMAGE_ASPECT_COLOR_BIT;
+		range.baseMipLevel   = 0;
+		range.levelCount     = VK_REMAINING_MIP_LEVELS;
+		range.baseArrayLayer = 0;
+		range.layerCount     = VK_REMAINING_ARRAY_LAYERS;
+
+		if (enable_dynamic)
+		{
+			vkb::insert_image_memory_barrier(draw_cmd_buffer,
+			                                 swapchain_buffers[i].image,
+			                                 0,
+			                                 VK_ACCESS_SHADER_WRITE_BIT,
+			                                 VK_IMAGE_LAYOUT_PRESENT_SRC_KHR,
+			                                 VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL,
+			                                 VK_PIPELINE_STAGE_TOP_OF_PIPE_BIT,
+			                                 VK_PIPELINE_STAGE_FRAGMENT_SHADER_BIT, range);
+
+			VkRenderingAttachmentInfoKHR color_attachment_info = vkb::initializers::rendering_attachment_info();
+			color_attachment_info.imageView                    = swapchain_buffers[i].view;        // color_attachment.image_view;
+			color_attachment_info.imageLayout                  = VK_IMAGE_LAYOUT_COLOR_ATTACHMENT_OPTIMAL;
+			color_attachment_info.resolveMode                  = VK_RESOLVE_MODE_NONE;
+			color_attachment_info.loadOp                       = VK_ATTACHMENT_LOAD_OP_CLEAR;
+			color_attachment_info.storeOp                      = VK_ATTACHMENT_STORE_OP_STORE;
+			color_attachment_info.clearValue                   = clear_values[0];
+
+			VkRenderingAttachmentInfoKHR depth_attachment_info = vkb::initializers::rendering_attachment_info();
+			depth_attachment_info.imageView                    = depth_stencil.view;
+			depth_attachment_info.imageLayout                  = VK_IMAGE_LAYOUT_DEPTH_ATTACHMENT_OPTIMAL;
+			depth_attachment_info.resolveMode                  = VK_RESOLVE_MODE_NONE;
+			depth_attachment_info.loadOp                       = VK_ATTACHMENT_LOAD_OP_CLEAR;
+			depth_attachment_info.storeOp                      = VK_ATTACHMENT_STORE_OP_STORE;
+			depth_attachment_info.clearValue                   = clear_values[1];
+
+			auto render_area               = VkRect2D{VkOffset2D{}, VkExtent2D{width, height}};
+			auto render_info               = vkb::initializers::rendering_info(render_area, 1, &color_attachment_info);
+			render_info.layerCount         = 1;
+			render_info.pDepthAttachment   = &depth_attachment_info;
+			render_info.pStencilAttachment = &depth_attachment_info;
```

Comment:
Also need pipeline-barrier for the depth/stencil buffer, syncing with use in previous frame. Just need one of them, either before or after rendering, to sync with previous/next frame respectively so there's no race.